streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -606,6 +606,25 @@ public void onAssignment(Assignment assignment) {
             }
         }
         metadataWithInternalTopics = Cluster.empty().withPartitions(topicToPartitionInfo);
+        updateSubscribedTopics(assignment);
+    }
+
+
+    private void updateSubscribedTopics(final Assignment assignment) {
```

Comment:
Yes.  This will cause a rebuild of the topology in those cases where all topics are already known. But thinking about it some more, IMHO it's worth the tradeoff for simplifying the logic.